### PR TITLE
add env switch. Closes #304

### LINKED ISSUE #304: Add env switching
Based on https://github.com/apex/apex/issues/253

### PR DESCRIPTION
**Pull request by mthenw**
I will add setting up other envs in separate PR :)

### CODE REVIEW

**Comment by mthenw** [comment]
@tj could you take a look?

**Comment by tj** [comment]
nice LGTM, maybe add the default to the flag as well. I wonder if we should use "stage" as well, instead of dev for default

**Comment by mthenw** [comment]
@tj or maybe there should be a question:

`Which environment do you want to setup (dev, test, stage, prod, none)?`

and until `none` it will be asking in a loop.

**Comment by tj** [comment]
hmm yea that sounds like a good plan. Maybe:

- `List the environments you would like: stage,prod` (something like that)
- `Which environment should be the default? stage`

maybe we store that in ./project.json?

**Comment by mthenw** [comment]
@tj updated

File: boot/boot.go [hunk 1]
```diff
@@ -109,31 +114,74 @@ func initProject(name, description, iamRole string) error {
...
+
+// setupEnv creates environment dir
+func setupEnv(env string) error {
+	logf("creating %s environment", env)
+
+	if err := os.MkdirAll(filepath.Join(infra.Dir, env), 0755); err != nil {
+		return err
+	}
+
+	maintf := filepath.Join(infra.Dir, "_env", "main.tf")
```
Comment (tj) [inline boot/boot.go]: what's this ./infrastructure/_env/main.tf for? (might be reading that wrong)

**Comment by tj** [comment]
cool looks good! Just wondering about that one _env dir

**Comment by mthenw** [comment]
What about "_env"? You want me to rename it?

File: boot/boot.go (hunk 1 above)
Comment (mthenw) [inline boot/boot.go]: it's just a placeholder :)

**Pull request merged**